### PR TITLE
Add pre-req libs for linux

### LINKED ISSUE #324: error failed to run custom build command for x11 v2.18.2
### Expected Behaviour
the example binary should build and display the example sky shader window.

### Example & Steps To Reproduce
1. clone the rust-gpu repo
```
git clone --recurse-submodules https://github.com/EmbarkStudios/rust-gpu
```
2. run the setup.sh
```
cd rust-gpu/
sh setup.sh
```
3. run the cargo run
```
cargo run --bin example-runner-wgpu --release
```
The error output follows:
```
error: failed to run custom build command for `x11 v2.18.2`

Caused by:
  process didn't exit successfully: `/home/davidm/rustplay/rust-gpu/target/release/build/x11-9180afe3ad5749c3/build-script-build` (exit code: 101)
  --- stdout
  cargo:rerun-if-env-changed=X11_NO_PKG_CONFIG
  cargo:rerun-if-env-changed=PKG_CONFIG
  cargo:rerun-if-env-changed=X11_STATIC
  cargo:rerun-if-env-changed=X11_DYNAMIC
  cargo:rerun-if-env-changed=PKG_CONFIG_ALL_STATIC
  cargo:rerun-if-env-changed=PKG_CONFIG_ALL_DYNAMIC
  cargo:rerun-if-env-changed=PKG_CONFIG_PATH_x86_64-unknown-linux-gnu
  cargo:rerun-if-env-changed=PKG_CONFIG_PATH_x86_64_unknown_linux_gnu
  cargo:rerun-if-env-changed=HOST_PKG_CONFIG_PATH
  cargo:rerun-if-env-changed=PKG_CONFIG_PATH
  cargo:rerun-if-env-changed=PKG_CONFIG_LIBDIR_x86_64-unknown-linux-gnu
  cargo:rerun-if-env-changed=PKG_CONFIG_LIBDIR_x86_64_unknown_linux_gnu
  cargo:rerun-if-env-changed=HOST_PKG_CONFIG_LIBDIR
  cargo:rerun-if-env-changed=PKG_CONFIG_LIBDIR
  cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR_x86_64-unknown-linux-gnu
  cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR_x86_64_unknown_linux_gnu
  cargo:rerun-if-env-changed=HOST_PKG_CONFIG_SYSROOT_DIR
  cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR

  --- stderr
  thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Failure { command: "\"pkg-config\" \"--libs\" \"--cflags\" \"x11\" \"x11 >= 1.4.99.1\"", output: Output { status: ExitStatus(ExitStatus(256)), stdout: "", stderr: "Package x11 was not found in the pkg-config search path.\nPerhaps you should add the directory containing `x11.pc\'\nto the PKG_CONFIG_PATH environment variable\nPackage \'x11\', required by \'virtual:world\', not found\nPackage \'x11\', required by \'virtual:world\', not found\n" } }', /home/davidm/.cargo/registry/src/github.com-1ecc6299db9ec823/x11-2.18.2/build.rs:36:67
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
warning: build failed, waiting for other jobs to finish...
error: build failed
```

### System Info
Fedora Rawhide
```
uname -a
Linux rawhide2020 5.10.0-0.rc6.20201202git509a15421674.91.fc34.x86_64 #1 SMP Wed Dec 2 16:26:23 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux
```

Rust Nightly
```
rustc -V
rustc 1.50.0-nightly (d9a105fdd 2020-11-23)
```

spirv-tools
```
spirv-tools-2020.5-2.20201031.gitf7da527.fc34.x86_64	API and commands for processing SPIR-V modules
spirv-tools-libs-2020.5-2.20201031.gitf7da527.fc34.x86_64	Library files for spirv-tools

spirv-val --version
SPIRV-Tools v2020.6 unknown hash, 2020-11-02T00:00:00
Targets:
  SPIR-V 1.0
  SPIR-V 1.1
  SPIR-V 1.2
  SPIR-V 1.3
  SPIR-V 1.4
  SPIR-V 1.5
  SPIR-V 1.2 (under OpenCL 2.2 Full Profile semantics)
  SPIR-V 1.0 (under Vulkan 1.0 semantics)
  SPIR-V 1.3 (under Vulkan 1.1 semantics)
  SPIR-V 1.4 (under Vulkan 1.1 semantics)
  SPIR-V 1.3 (under WIP WebGPU semantics)
```

### Resolution to above problem
Install remmina and it will install all dependant x11 packages.  Also install libX11-devel for the necessary header files the rust/spir-v compiler will need.
```
dnf install remmina libX11-devel 
```

### LINKED ISSUE #325: error failed to run custom build command for spirv-tools-sys v0.1.1
### Expected Behaviour
Build and display the example sky shader window

### Example & Steps To Reproduce
1. clone the rust-gpu repo
```
git clone --recurse-submodules https://github.com/EmbarkStudios/rust-gpu
```
2. run the setup.sh
```
cd rust-gpu/
sh setup.sh
```
3. install remmina and x11 devel
```
dnf install remmina libX11-devel 
```
4. run the cargo run
```
cargo run --bin example-runner-wgpu --release
```

The error output follows:
```
error: failed to run custom build command for `spirv-tools-sys v0.1.1`

Caused by:
  process didn't exit successfully: `/home/davidm/rustplay/rust-gpu/target/release/build/spirv-tools-sys-64c6ee0db547dd5f/build-script-build` (exit code: 1)
  --- stdout
  OPT_LEVEL = Some("0")
  TARGET = Some("x86_64-unknown-linux-gnu")
  HOST = Some("x86_64-unknown-linux-gnu")
  CC_x86_64-unknown-linux-gnu = None
  CC_x86_64_unknown_linux_gnu = None
  HOST_CC = None
  CC = None
  CFLAGS_x86_64-unknown-linux-gnu = None
  CFLAGS_x86_64_unknown_linux_gnu = None
  HOST_CFLAGS = None
  CFLAGS = None
  CRATE_CC_NO_DEFAULTS = None
  DEBUG = Some("false")
  CARGO_CFG_TARGET_FEATURE = Some("fxsr,sse,sse2")
  CXX_x86_64-unknown-linux-gnu = None
  CXX_x86_64_unknown_linux_gnu = None
  HOST_CXX = None
  CXX = None
  CXXFLAGS_x86_64-unknown-linux-gnu = None
  CXXFLAGS_x86_64_unknown_linux_gnu = None
  HOST_CXXFLAGS = None
  CXXFLAGS = None
  running: "c++" "-O0" "-ffunction-sections" "-fdata-sections" "-fPIC" "-m64" "-I" "spirv-tools/" "-I" "spirv-tools/include" "-I" "generated/" "-I" "spirv-headers/include" "-I" "spirv-headers/include/spirv/unified1" "-Wall" "-Wextra" "-Wall" "-Wextra" "-Wnon-virtual-dtor" "-Wno-missing-field-initializers" "-Werror" "-std=c++11" "-fno-exceptions" "-fno-rtti" "-Wno-long-long" "-Wshadow" "-Wundef" "-Wconversion" "-Wno-sign-conversion" "-std=gnu++11" "-DSPIRV_CHECK_CONTEXT" "-DSPIRV_LINUX" "-o" "/home/davidm/rustplay/rust-gpu/target/release/build/spirv-tools-sys-35722c24cafd7636/out/spirv-tools/source/util/bit_vector.o" "-c" "spirv-tools/source/util/bit_vector.cpp"
  running: "c++" "-O0" "-ffunction-sections" "-fdata-sections" "-fPIC" "-m64" "-I" "spirv-tools/" "-I" "spirv-tools/include" "-I" "generated/" "-I" "spirv-headers/include" "-I" "spirv-headers/include/spirv/unified1" "-Wall" "-Wextra" "-Wall" "-Wextra" "-Wnon-virtual-dtor" "-Wno-missing-field-initializers" "-Werror" "-std=c++11" "-fno-exceptions" "-fno-rtti" "-Wno-long-long" "-Wshadow" "-Wundef" "-Wconversion" "-Wno-sign-conversion" "-std=gnu++11" "-DSPIRV_CHECK_CONTEXT" "-DSPIRV_LINUX" "-o" "/home/davidm/rustplay/rust-gpu/target/release/build/spirv-tools-sys-35722c24cafd7636/out/spirv-tools/source/util/parse_number.o" "-c" "spirv-tools/source/util/parse_number.cpp"

  --- stderr


  error occurred: Failed to find tool. Is `c++` installed?


warning: build failed, waiting for other jobs to finish...
error: build failed

```

### Environment
Fedora Rawhide with the following os packages for spirv-tools installed.  I don't know if installing this was a mistake or not, but spirv-val doesn't seem to exist in any of the crates that cargo fetched to build the example.
```
 spirv-tools-2020.5-2.20201031.gitf7da527.fc34.x86_64	API and commands for processing SPIR-V modules
 spirv-tools-libs-2020.5-2.20201031.gitf7da527.fc34.x86_64	Library files for spirv-tools

spirv-val --version
SPIRV-Tools v2020.6 unknown hash, 2020-11-02T00:00:00
Targets:
  SPIR-V 1.0
  SPIR-V 1.1
  SPIR-V 1.2
  SPIR-V 1.3
  SPIR-V 1.4
  SPIR-V 1.5
  SPIR-V 1.2 (under OpenCL 2.2 Full Profile semantics)
  SPIR-V 1.0 (under Vulkan 1.0 semantics)
  SPIR-V 1.3 (under Vulkan 1.1 semantics)
  SPIR-V 1.4 (under Vulkan 1.1 semantics)
  SPIR-V 1.3 (under WIP WebGPU semantics)
```

### Resolution
```
dnf install gcc-c++
```

### LINKED ISSUE #326: error failed to run custom build command for xkbcommon-sys v0.7.4
### Expected Behaviour
Build the binary and run the output window.

### Example & Steps To Reproduce
1. clone the rust-gpu repo
```
git clone --recurse-submodules https://github.com/EmbarkStudios/rust-gpu
```
2. run the setup.sh
```
cd rust-gpu/
sh setup.sh
```
3. run the cargo run
```
cargo run --bin example-runner-cpu --release
```

The error output follows:
```
error: failed to run custom build command for `xkbcommon-sys v0.7.4`

Caused by:
  process didn't exit successfully: `/home/davidm/rustplay/rust-gpu/target/release/build/xkbcommon-sys-d4c1ad9711ce8555/build-script-build` (exit code: 101)
  --- stdout
  cargo:rerun-if-env-changed=XKBCOMMON_NO_PKG_CONFIG
  cargo:rerun-if-env-changed=PKG_CONFIG
  cargo:rerun-if-env-changed=PKG_CONFIG_PATH_x86_64-unknown-linux-gnu
  cargo:rerun-if-env-changed=PKG_CONFIG_PATH_x86_64_unknown_linux_gnu
  cargo:rerun-if-env-changed=HOST_PKG_CONFIG_PATH
  cargo:rerun-if-env-changed=PKG_CONFIG_PATH
  cargo:rerun-if-env-changed=PKG_CONFIG_LIBDIR_x86_64-unknown-linux-gnu
  cargo:rerun-if-env-changed=PKG_CONFIG_LIBDIR_x86_64_unknown_linux_gnu
  cargo:rerun-if-env-changed=HOST_PKG_CONFIG_LIBDIR
  cargo:rerun-if-env-changed=PKG_CONFIG_LIBDIR
  cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR_x86_64-unknown-linux-gnu
  cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR_x86_64_unknown_linux_gnu
  cargo:rerun-if-env-changed=HOST_PKG_CONFIG_SYSROOT_DIR
  cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR

  --- stderr
  thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Failure { command: "\"pkg-config\" \"--libs\" \"--cflags\" \"xkbcommon\"", output: Output { status: ExitStatus(ExitStatus(256)), stdout: "", stderr: "Package xkbcommon was not found in the pkg-config search path.\nPerhaps you should add the directory containing `xkbcommon.pc\'\nto the PKG_CONFIG_PATH environment variable\nPackage \'xkbcommon\', required by \'virtual:world\', not found\n" } }', /home/davidm/.cargo/registry/src/github.com-1ecc6299db9ec823/xkbcommon-sys-0.7.4/build.rs:43:14
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
warning: build failed, waiting for other jobs to finish...
error: build failed
```

### Resolution
Install the respective development library and headers:
```
dnf install libxkbcommon-x11-devel
```

### PR DESCRIPTION
**Pull request by Jasper-Bekkers**
Fixes #324
Fixes #325
Fixes #326

Add a small blurb for our linux users about required packages.

### CODE REVIEW

**Comment by khyperia** [comment]
Could we look up package names for non-Fedora linux distros as well?

**Comment by khyperia** [comment]
The issues mention `libX11-devel`, `libxkbcommon-x11-devel`, and `gcc-c++`. Why aren't the (ubuntu-equivalents) second two listed in this PR?

**Comment by Jasper-Bekkers** [comment]
I'm not a native linux user so I have no idea what they are, there where the ones I needed to install on my WSL setup.

File: docs/src/building-rust-gpu.md [hunk 1]
```diff
@@ -19,6 +19,14 @@
 
     This will build `rustc_codegen_spirv`, the compiler, then use that compiler to build [`sky-shader`](examples/shaders/sky-shader) into a SPIR-V module, then finally, build a `wgpu` sample app (modified from [`wgpu`'s examples](https://github.com/gfx-rs/wgpu-rs/tree/master/examples/hello-triangle)) using the built SPIR-V module to display the shader in a window.
 
+## Prerequiste packages recommended to install before building Rust-GPU
```
Comment (XAMPPRocky) [inline docs/src/building-rust-gpu.md:22]:
```suggestion
## Prerequisite packages recommended to install before building Rust-GPU
```

**Comment by khyperia** [comment]
> I'm not a native linux user so I have no idea what they are, there where the ones I needed to install on my WSL setup.

They may have already been installed, or WSL may differ from actual ubuntu here (especially relevant is the lack of X11-by-default in WSL). Please do investigate if they're needed or not before merging - they were reported as needed for a reason~

**Comment by khyperia** [comment]
@Jasper-Bekkers do you plan on following up with this, or should we just close this?

**Pull request closed without merging**